src/deepsparse/schedulers/continuous_batching_scheduler.py

```diff
@@ -167,16 +167,30 @@ def add_engine_operator(
                 continue  # already added
 
             override_model_path = None
-            # text generation/NLEngineOperator specific; could add generic method
-            # for all engine_operators, if desired
-            if hasattr(engine_operator, "override_model_inputs"):
-                override_model_path = engine_operator.override_model_inputs(
-                    model_path=engine_operator.model_path, batch_size=batch_size
+            kwargs = {}
+
+            if engine_operator.__class__.__name__ == "NLEngineOperator":
```

Comment:
can any of this logic be moved into the operators itself? just worried about tying in the operator class too much

Reply:
so all of this is needed for engine creation which happens in the operator's init function for batch_size 1. We could move these functions outside of the engine init, override the engine'e `create_engine` method, and add these steps to it.

Reply:
Done.